packages/shared-internals/src/colocation.ts

```diff
@@ -40,9 +41,18 @@ function correspondingJSExists(id: string): boolean {
   return ['js', 'ts'].some(ext => existsSync(id.slice(0, -3) + ext));
 }
 
-function isInComponents(id: string, packageCache: Pick<PackageCache, 'ownerOfFile'>) {
+export function isInComponents(id: string, packageCache: Pick<PackageCache, 'ownerOfFile'>) {
   const pkg = packageCache.ownerOfFile(id);
-  return pkg?.isV2App() && id.slice(pkg?.root.length).split(sep).join('/').startsWith('/components');
+  if (!pkg?.isV2App()) {
+    return false;
+  }
+
+  let tryResolve = resolveExports(pkg.packageJSON, './components', {
```

Comment:
We need to identify where `my-app/components/` has been mapped to via the package.json exports, instead of just assuming it will be `./app/components`.

Reply:
nice one, great point! thanks for adding this 👍